hal/halx86/apic/apictimer.c

```diff
@@ -63,22 +69,75 @@ VOID
 NTAPI
 HalStartProfileInterrupt(IN KPROFILE_SOURCE ProfileSource)
 {
-    UNIMPLEMENTED;
-    return;
+	LVT_REGISTER LvtEntry;
+
+	/* Only handle ProfileTime */
+	if (ProfileSource == ProfileTime)
+	{
+		/* OK, we are profiling now */
+		HalIsProfiling = TRUE;
+
+		/* Set interrupt interval */
+		ApicWrite(APIC_TICR, PCR->HalReserved[0]);
```

Comment:
Please always use KeGetPcr() instead of using the PCR define directly.

Reply:
OK, will fix this